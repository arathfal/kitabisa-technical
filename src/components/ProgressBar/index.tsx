'use client'

import { useMemo } from 'react'

import classNames from '@/lib/classNames'

type ProgressBarProps = {
  value?: number
}

export default function ProgressBar({ value = 0 }: ProgressBarProps) {
  const { width, isFull } = useMemo(() => {
    return {
      width: `${value}%`,
      isFull: value === 100
    }
  }, [value])

  return (
    <div className="relative w-full overflow-hidden rounded-full bg-slate-300">
      <div
        className={classNames('h-2 text-right', isFull ? 'bg-pink-500' : 'bg-gray-500')}
        style={{ width }}
      />
    </div>
  )
}
