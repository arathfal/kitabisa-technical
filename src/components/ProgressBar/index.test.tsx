import { cleanup, render } from '@testing-library/react'

import ProgressBar from '.'

const renderProgressBar = (value = 0) => ({
  ...render(<ProgressBar value={value} />)
})

describe('ProgressBar Component', () => {
  afterEach(cleanup)

  test('should render with text value 0', () => {
    const { asFragment } = renderProgressBar()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render with text value 100', () => {
    const { asFragment } = renderProgressBar(100)
    expect(asFragment()).toMatchSnapshot()
  })
})
