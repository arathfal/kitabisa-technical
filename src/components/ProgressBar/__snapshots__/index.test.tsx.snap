// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component should render with text value 0 1`] = `
<DocumentFragment>
  <div
    class="relative w-full overflow-hidden rounded-full bg-slate-300"
  >
    <div
      class="h-2 text-right bg-gray-500"
      style="width: 0%;"
    />
  </div>
</DocumentFragment>
`;

exports[`ProgressBar Component should render with text value 100 1`] = `
<DocumentFragment>
  <div
    class="relative w-full overflow-hidden rounded-full bg-slate-300"
  >
    <div
      class="h-2 text-right bg-pink-500"
      style="width: 100%;"
    />
  </div>
</DocumentFragment>
`;
